src/parallel/executor.cpp

```diff
@@ -675,4 +677,168 @@ unique_ptr<DataChunk> Executor::FetchChunk() {
 	return chunk;
 }
 
+idx_t Executor::CountChunk() {
```

Comment:
Why are all these changes being made to the executor? This is out of scope for this PR and should not be necessary. Please revert.

Reply:
This is necessary, because we must know all source count as the result of SQL execution without "limit %" before the entire SQL execution. This code doesn't affect other parts, only affects limit percent operator.

Reply:
This approach is very unmaintainable, since the entire query execution logic is now duplicated for the sole purpose of a single operator. It does not seem efficient either, since the entire query is now run multiple times.

The limit % operator is not special, it can be implemented without any changes to the executor.

Reply:
Okay, I will find another way without using of executor